hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -86,6 +86,8 @@ public DynamicTableSource createDynamicTableSource(Context context) {
     setupTableOptions(conf.getString(FlinkOptions.PATH), conf);
     ResolvedSchema schema = context.getCatalogTable().getResolvedSchema();
     setupConfOptions(conf, context.getObjectIdentifier(), context.getCatalogTable(), schema);
+    HoodieTableMetaClient metaClient = StreamerUtil.metaClientForReader(conf, HadoopConfigurations.getHadoopConf(conf));
+    SanityChecksUtil.sanitCheck(conf, schema, metaClient);
```

Comment:
Move the instantiation of meta client into the SanityCheckUtil. And can we rename the class to `SanityChecks`